File: JavaScript/01.JavaScript-Basic-February-2021/10.whileLoop-Exersice/04.walking.js
function walking(input) {
    let steps = 0;
    let index = 0;
    let command = input[index];
    index++;

    while (steps <= 10000) {
        if (command === "Going home") {
            steps += Number(input[index]);
            index++;
            break;
        }

        let tempStep = Number(command);
        steps += tempStep;

        command = input[index];
        index++
    }

    if (steps >= 10000) {
        let diff = steps - 10000;
        console.log(`Goal reached! Good job!`);
        console.log(`${diff} steps over the goal!`)
    } else {
        let diff = 10000 - steps;
        console.log(`${diff} more steps to reach goal.`)
    }
}

walking(["1000", "1500", "2000", "6500"]);
walking(["1000", "1500", "2000", "6500"]);
walking(["1500", "3000", "250", "1548", "2000", "Going home", "2000"]);
walking(["125", "250", "4000", "30", "2678", "4682"]);

File: Python/01.Python-Basic-April-2019/09.while_loop_exercise/04.walking.py
steps_counter = 0

while True:
    current_steps = input()
    if current_steps == 'Going home':
        current_steps = int(input())

        steps_counter += current_steps

        if steps_counter >= 10000:
            print("Goal reached! Good job!")
            break
        else:
            print(f'{10000 - steps_counter} more steps to reach goal.')
            break
    else:
        current_steps = int(current_steps)

    steps_counter += current_steps

    if steps_counter >= 10000:
        print("Goal reached! Good job!")
        break